src/main/java/com/staff_management_system/com/database_connection_package/HourlyEmployeeDataAccess.java

```diff
@@ -0,0 +1,266 @@
+    String password;
+    String firstName;
+    String lastName;
+    String role;
+    Date startDate;
+    String sin;
+    long phoneNumber;
+    String emailAddress;
+    String address;
+    float hourly;
+    boolean enabled;
+    String department;
+    // retrieve employee data
+    /**
+     * variables for the database connection
+     */
+    private HourlyEmployee anHourlyEmployee = null;
+    private Connection aConnection;
+    private Statement aStatement;
+
+    /**
+     * @param c establish the database connection
+     */
+
+    public void initialize(Connection c) {
+        try {
+            aConnection = c;
+            aStatement = aConnection.createStatement();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * close the database connection
+     */
+    public void terminate() {
+        try {
+            aStatement.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    /**
+     * retrieve the database connection
+     *
+     * @param id the id for employee
+     * @return anHourlyEmployee
+     * @throws NotFoundException, InvalidNameException, InvalidIdException, InvalidPasswordException
+     */
+    public HourlyEmployee retrieve(int id) throws NotFoundException, InvalidNameException, InvalidIdException, InvalidPasswordException {
+        anHourlyEmployee = null;
+        try {
+            aStatement = aConnection.createStatement();
+            // execute the SQL query statement
+            ResultSet rs = aStatement.executeQuery("SELECT * FROM tblhourlyemployee WHERE id=" + id);
+
+            // next method sets cursor & returns true if there is data
+
+            if (rs.next()) { // extract the data
+                anHourlyEmployee.setId(rs.getInt("id"));
+                anHourlyEmployee.setPassword(rs.getString("password"));
+                anHourlyEmployee.setFirstName(rs.getString("firstname"));
+                anHourlyEmployee.setLastName(rs.getString("lastname"));
+                anHourlyEmployee.setRole(rs.getString("role"));
+                anHourlyEmployee.setStartDate(rs.getDate("startdate"));
+                anHourlyEmployee.setSin(rs.getString("sin"));
+                anHourlyEmployee.setPhoneNumber(rs.getLong("phonenumber"));
+                anHourlyEmployee.setEmailAddress(rs.getString("emailaddress"));
+                anHourlyEmployee.setAddress(rs.getString("address"));
+                anHourlyEmployee.setHourly(rs.getFloat("hourly"));
+                anHourlyEmployee.setEnabled(rs.getBoolean("enabled"));
+                anHourlyEmployee.setDepartment(rs.getString("department"));
+
+
+                try {
+                    anHourlyEmployee = new HourlyEmployee(id, password, firstName, lastName, role, startDate, sin, phoneNumber,
+                            emailAddress, address, hourly, enabled, department);
+                } catch (InvalidEmployeeDataException e) {
```

Comment:
you will not get the exception that the employee does not exist here .. you will get it when you call the database trying to retrieve employee with non-existent ID

Reply:
I will change that.